migrations/20210612_initial.sql

```diff
@@ -0,0 +1,104 @@
+CREATE TABLE profiles (
+  profile_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL,
+  properties TEXT NOT NULL -- JSON with user or contact profile
+);
+
+CREATE TABLE known_servers(
+  host TEXT NOT NULL,
+  port TEXT NOT NULL,
+  key_hash BLOB,
+  PRIMARY KEY (host, port)
+) WITHOUT ROWID;
+
+CREATE TABLE contacts (
+  contact_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL, -- display name set by local user
+  properties TEXT NOT NULL, -- JSON set by local user
+  profile_id INTEGER REFERENCES profiles, -- sent by remote contact
```

Comment:
if a profile is sent by remote contact, how is it that a profile can be referenced by multiple contacts? shouldn't it be vice versa?

if it's 1 to 1 `profile`'s `properties` could just be stored on `contact` as `external_properties`

Reply:
given that I removed users there is no distinction now between contact and profile, probably best to just merge the fields into one table. It's 1:1 relationship...